### PR TITLE
Possible bug in auditSync

### PR DESCRIPTION
**Pull request by tricki**

### CODE REVIEW

File: tests/Functional/AuditingTest.php [hunk 1]
```diff
@@ -620,13 +620,16 @@ public function itWillAuditSync()
...
+        $audit = $article->audits()->where('event', 'sync')->latest()->first();
         $no_of_audits_after = Audit::where('auditable_type', Article::class)->count();
         $categoryAfter = $article->categories()->first()->getKey();
 
         $this->assertSame($firstCategory->getKey(), $categoryBefore);
         $this->assertSame($secondCategory->getKey(), $categoryAfter);
         $this->assertNotSame($categoryBefore, $categoryAfter);
         $this->assertGreaterThan($no_of_audits_before, $no_of_audits_after);
+        $this->assertCount(1, $audit->old_values['categories']);
+        $this->assertCount(2, $audit->new_values['categories']);
```
Comment (erikn69) [inline tests/Functional/AuditingTest.php:632]: `sync` is expected to delete the entire relationship and re-attach it

https://laravel.com/docs/11.x/eloquent-relationships#syncing-associations

So after sync, `old_values` has 1, and `new_values` has 1

Comment (tricki) [reply tests/Functional/AuditingTest.php:662]: Yeah, I just saw that. I've pushed a fix and a new test which illustrates my problem better. I wasn't expecting you to be this fast 😅
Here's the corresponding issue with more explanation: https://github.com/owen-it/laravel-auditing/issues/931

File: tests/Functional/AuditingTest.php [hunk 2]
```diff
@@ -620,13 +620,46 @@ public function itWillAuditSync()
...
+        $firstCategory = factory(Category::class)->create();
+        $secondCategory = factory(Category::class)->create();
+        $thirdCategory = factory(Category::class)->create();
+        $article = factory(Article::class)->create();
+
+        $article->categories()->attach($firstCategory);
+
+        $no_of_audits_before = Audit::where('auditable_type', Article::class)->count();
+
+        $article->auditSync('categories', [$secondCategory->getKey(), $thirdCategory->getKey()]);
```
Comment (willpower232) [inline tests/Functional/AuditingTest.php:650]: are you expecting this to **add** two categories or **set** the whole relation to just these two categories?

File: tests/Functional/AuditingTest.php (hunk 1 above)
Comment (erikn69) [reply tests/Functional/AuditingTest.php:662]: the change I made is how I understood your problem

File: tests/Functional/AuditingTest.php (hunk 2 above)
Comment (tricki) [reply tests/Functional/AuditingTest.php:650]: To **set** it to the two categories. I'm also expecting that "old" contains the first category (i.e. the same as `$article->categories` before the sync), and "new" the second and third (i.e. the same as `$article->categories` after the sync).

Comment (willpower232) [reply tests/Functional/AuditingTest.php:650]: Makes sense thanks, looks like @erikn69 tweaked the test so it fails now so hopefully is on the case 🙏

**Comment by parallels999** [comment]
It seems it was a misunderstanding: https://github.com/owen-it/laravel-auditing/issues/931#issuecomment-2126371800

**Pull request closed without merging**